optimum/exporters/onnx/config.py

```diff
@@ -164,10 +164,7 @@ def inputs(self) -> Dict[str, Dict[int, str]]:
         # generating wrong position_ids in the model itself:
         # https://github.com/huggingface/transformers/blob/v4.33.1/src/transformers/models/gpt2/modeling_gpt2.py#L802
         if not self.no_position_ids and self.task == "text-generation":
-            if self.use_past_in_inputs:
-                common_inputs["position_ids"] = {0: "batch_size"}
-            else:
-                common_inputs["position_ids"] = {0: "batch_size", 1: "sequence_length"}
+            common_inputs["position_ids"] = {0: "batch_size", 1: "sequence_length"}
```

Comment:
Why?

Reply:
The sequence length of the `position_ids` can vary and is not necessarily 1 anymore (first generation)

Reply:
Isn't this "breaking" the legacy export? Breaking in the sense that previously the with past decoder was hard-coding `1` shape and now would not. It is not a very minor issue anyway given that the merged decoder grab the shape from the `decoder_model.onnx`.

Reply:
discussed internally and resolved